sklearn/utils/validation.py

```diff
@@ -92,7 +92,7 @@ def _is_arraylike(x):
 
 def _num_samples(x):
     """Return number of samples in array-like x."""
-    if hasattr(x, 'fit'):
+    if hasattr(x, 'fit') and hasattr(x.fit, '__call__'):
```

Comment:
That callable has been deprecated in some versions of python - 3.0 and 3.1 I think - I thought this would be more backwards compatible.